### DISCUSSION TITLE
Can't access sites via domain only via ip

### DISCUSSION THREAD

**Discussion by countradooku** [OpenVPN/openvpn3-linux#250] [Q&A - Our support arena]
Hello

After coinnecting to vpn via openvpn3 session-start successfully i can access the intranet websites only via IP but not via domain. On windows it works perfectly but her eon linux it doesn't. can someone help me?

**Comment by dsommers** [comment]
Can you please provide some details about your setup?  Which Linux distribution and version? How do you start the VPN session?

And can you run these command as `root` and provide the output?

     # openvpn3-admin version --services
     # openvpn3-admin init-config

(The `init-config` will not change anything on your system when run like this)

**Comment by countradooku** [comment]
Hello @dsommers 

I use ubuntu linux 23.10
![image](https://github.com/OpenVPN/openvpn3-linux/assets/52667211/d2de70cb-665c-4777-85c5-14df4f8bb0f6)
![image](https://github.com/OpenVPN/openvpn3-linux/assets/52667211/b3ba5669-15c4-45a4-be10-705a1fdcaaa4)

i start the session like this 
openvpn3 session-start --config myconfig

**Reply by dsommers** [reply]
Okay, first - Ubuntu 23.10 is potentially unstable with OpenVPN 3 Linux v21.  Please install this development version of OpenVPN 3 Linux, using this PPA repository: https://github.com/OpenVPN/openvpn3-linux/issues/171#issuecomment-1976941394

So to the output from the `init-config` command ... The "Network Configuration" result is a bit unexpected.  Can you verify that your host really uses `/etc/resolv.conf` and not  the `systemd-resolved` service?  If your host uses `/etc/resolv.conf`, you can tell OpenVPN 3 Linux to use that too by running this command:

     # openvpn3-admin netcfg-service --config-unset systemd-resolved   # !! This command may fail; that's okay
     # openvpn3-admin netcfg-service --config-set resolv-conf /etc/resolv.conf

**Reply by countradooku** [reply]
ok i run the commands

**Reply by countradooku** [reply]
![image](https://github.com/OpenVPN/openvpn3-linux/assets/52667211/20989aa1-ed3f-4358-bccc-687f239ec6dd)
this is my output now

**Reply by countradooku** [reply]
@dsommers still doesn't work

**Reply by dsommers** [reply]
What is the content of `/etc/resolv.conf` before and after the VPN session has started?

**Comment by countradooku** [comment]
![image](https://github.com/OpenVPN/openvpn3-linux/assets/52667211/5591ccfd-5f6b-49dd-abec-6d6bc0d4d830)

and after it is unchanged

**Reply by countradooku** [reply]
do it added the domains i guess. but still cant access them

**Reply by dsommers** [reply]
I also asked for `/etc/resolv.conf` **before** the VPN session started.  You will need to share the complete content.  This file does not contain anything security sensitive at all (IP addresses are not secrets).

And instead of a "screenshot", please just copy paste the content from your terminal.  Just use the ["Quoting code" trick](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) in github.

**Reply by countradooku** [reply]
That was the content before starting the session

**Reply by dsommers** [reply]
The opening line "Generated by OpenVPN 3 Linux" already then indicates an issue on your host .... do you have an `/etc/resolv.conf.ovpn3bak` file present?  Does that contain the same?  If that is truly what should be the original, move that to `/etc/resolv.conf`.  Otherwise, you need to reconfigure this file to the default DNS servers you use before you start a VPN tunnel.

**Comment by countradooku** [comment]
![image](https://github.com/OpenVPN/openvpn3-linux/assets/52667211/001c712e-b474-41f0-b553-232bf5b73848)

**Comment by countradooku** [comment]
```
# This is /run/systemd/resolve/resolv.conf managed by man:systemd-resolved(8).
# Do not edit.
#
# This file might be symlinked as /etc/resolv.conf. If you're looking at
# /etc/resolv.conf and seeing this text, you have followed the symlink.
#
# This is a dynamic resolv.conf file for connecting local clients directly to
# all known uplink DNS servers. This file lists all configured search domains.
#
# Third party programs should typically not access this file directly, but only
# through the symlink at /etc/resolv.conf. To manage man:resolv.conf(5) in a
# different way, replace this symlink by a static file or a different symlink.
#
# See man:systemd-resolved.service(8) for details about the supported modes of
# operation for /etc/resolv.conf.

nameserver 192.168.1.1
search .
```

this is the content for /etc/resolv.conf.ovpn3bak

**Reply by dsommers** [reply]
Okay .... it looks like your whole resolver setup is in a bad shape.  That should be a symlink to `/run/systemd/resolve/resolv.conf` .... but that's again not the recommended way how to use `systemd-resolved`.  At the same time, I also know that Ubuntu has it's own way of doing stuff; it does ship with a half-baked `systemd-resolved` setup - and I'm not up-to-date on what it does on Ubuntu 23.10.

I would need to setup a 23.10 host somewhere and poke at it more carefully to better propose a good way forward.  I won't be able to do that today, unfortunately (calendar is filled up today).

**Reply by countradooku** [reply]
@dsommers why i get this?
![image](https://github.com/OpenVPN/openvpn3-linux/assets/52667211/4e5b2338-f19b-4dfb-a9ba-7473dd7a8063)

**Reply by dsommers** [reply]
You need to first unset the resolv-conf setting:

     sudo openvpn3-admin netcfg-service --config-unset resolv-conf

The `resolv-conf` and `systemd-resolved` settings are exclusive, you can only use one of them - not both at the same time.
